server/message_queue_service.py

```diff
@@ -31,12 +37,18 @@ def __init__(self) -> None:
         config.register_callback("MQ_PORT", self.reconnect)
 
     async def initialize(self) -> None:
-        if self._connection is not None:
-            return
+        async with self._initialization_lock:
+            if self._connection is not None:
```

Comment:
Should we be checking `is_ready` here instead?

Reply:
Yes that seems better. TBH one could probably also replace the flag by `self._connection is not None` checks everywhere else, but having the flag feels more readable.